File: uri/1164/1164.cpp
#include <cstdio>
#include <cmath>

int main(){
    int x, n, s;
    scanf("%d", &x);
    while(x--){
        scanf("%d", &n);
        s = 0;
        for (int i = 1; i <= n / 2; i++){
            if (n % i == 0){
                s += i;
            }
        }
        if (s == n)
            printf("%d eh perfeito\n", n);
        else
            printf("%d nao eh perfeito\n", n);
    }
    return 0;
}

File: solutions/beecrowd/1164/1164.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

x = int(input())

for _ in range(x):
    n = int(input())

    s = 0

    for i in range(1, n // 2 + 1):
        if n % i == 0:
            s += i

    if s == n:
        output(f'{n} eh perfeito\n')
    else:
        output(f'{n} nao eh perfeito\n')
